### ISSUE TITLE
refine source definition file format

### ISSUE THREAD

**Issue by jdpye** [ocean-tracking-network/rt-sat-to-obis#2]
I'm thinking of a script that can handle multiple data sources, so I'm looking at using YAML files to define what each data source is, and  where their source fields are. Then, each source file would have URLs to the appropriate source data for telemetry (here meaning 'location data from the instrument'), tag attachment to individuals, and project-level metadata. 

 There's probably a lot I'm missing but I think this is a suitable skeleton to start from! See the strawman in https://github.com/ocean-tracking-network/rt-sat-to-obis/blob/main/sources/imos_ct180.yml

**Comment by MathewBiddle** [comment]
Would it be of interest to use [Frictionless Data Packages (FDP)](https://specs.frictionlessdata.io/guides/data-package/#data-package) here?

FDP spec: https://specs.frictionlessdata.io/data-package/

There is also the Library of Congress BagIt specification: https://blogs.loc.gov/thesignal/2019/04/bagit-at-the-library-of-congress/

**Comment by jdpye** [comment]
I'm not sure a FDP or other hard-standard for a between-products data object is useful, but maybe at the point that you foresee a fork in the road, where an intermediary data product we are making is intended to be used in creating multiple data products downstream, maybe then it becomes more useful? Pitch me on it if so, let me know what is helpful about it.

**Comment by MathewBiddle** [comment]
Makes sense. Just trying to not invent a new standard if others already exist. YAML is widely used and easy to work with, so lets see what we get.

**Comment by jdpye** [comment]
Coming around to there potentially being a use for FDP to map column names that are different into a standard vocab, but it's a more complicated way of doing so than just a standard mapping file.

ArgosQC is implementing a config file for the higher-order variables within a certain campaign, so it's possible that a mapping file or a 'driver' file for the instrument data could be defined in that config file and implemented when running the QC periodically.

**Comment by jdpye** [comment]
Operational needs seemed to suggest that a config_[CID].json file within each of the project folders would be able to drive the modelling and the publishing processes. There are example files in the IMOS folders now, with sections within the JSON for everything we're looking for, from model parameters to the institutional metadata template, to the location of contact metadata reference files.
